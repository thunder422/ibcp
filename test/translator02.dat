#################################################
#  Translator Test #2: Data Types (2010-04-28)  #
#################################################

Z = not A < 5.0 = B > 2.0
Z% = not A% < 5 and B% > 2 or C% < 1 and D% <> 2
Z$=STR$(VAL("1.23"))
Z$=MID$(A$+B$,(C%+5)*D%,4)+" Test"
# 2010-05-09: corrected expression
Z% = (a$ = "this" + "test")
Z = A% + B
Z = A + B%
Z% = A% + B%
Z = A% + B% + C
Z = A% - B
Z = A - B%
Z% = A% - B%
Z = A% - B% * C
Z = ABS(A%) + SGN(B%) + ABS(C) * SGN(D)
Z = ABS(-A%) + SGN(-B%) + ABS(-C) * SGN(-D)
Z$ = STR$(A) + STR$(A%)
Z = A% + B * C ^ D

# begin of error tests
Z = A$ + B
Z = A$ + B%
Z = A + B$
Z = A% + B$
Z$ = A$ - B$
Z$ = MID$(A$,B$,1)
Z$ = MID$(A$,2,C$)
Z$ = MID$(A,2,3)
Z$ = MID$(A$,2,3) + B
Z = B + MID$(A$,2,3)
Z = A + (B$ + C$)
