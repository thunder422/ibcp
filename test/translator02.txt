
Input: Z = not A < 5.0 = B > 2.0
Output: Z<ref> A 5.0 < B 2.0 > =% NOT CvtDbl =Assign 

Input: Z% = not A% < 5 and B% > 2 or C% < 1 and D% <> 2
Output: Z%<ref> A% 5% <% NOT B% 2% >% AND C% 1% <% D% 2% <>% AND OR =Assign% 

Input: Z$=STR$(VAL("1.23"))
Output: Z$<ref> "1.23" VAL( STR$( =Assign$ 

Input: Z$=MID$(A$+B$,(C%+5)*D%,4)+" Test"
Output: Z$<ref> A$ B$ +$ C% 5% +% D% *% 4% MID$(3 " Test" +$ =Assign$ 

Input: Z% = (a$ = "this" + "test")
Output: Z%<ref> a$ "this" "test" +$ =$')' =Assign% 

Input: Z = A% + B
Output: Z<ref> A% B +%1 =Assign 

Input: Z = A + B%
Output: Z<ref> A B% +%2 =Assign 

Input: Z% = A% + B%
Output: Z%<ref> A% B% +% =Assign% 

Input: Z = A% + B% + C
Output: Z<ref> A% B% +% C +%1 =Assign 

Input: Z = A% - B
Output: Z<ref> A% B -%1 =Assign 

Input: Z = A - B%
Output: Z<ref> A B% -%2 =Assign 

Input: Z% = A% - B%
Output: Z%<ref> A% B% -% =Assign% 

Input: Z = A% - B% * C
Output: Z<ref> A% B% C *%1 -%1 =Assign 

Input: Z = ABS(A%) + SGN(B%) + ABS(C) * SGN(D)
Output: Z<ref> A% ABS(% B% SGN(% +% C ABS( D SGN( * +%1 =Assign 

Input: Z = ABS(-A%) + SGN(-B%) + ABS(-C) * SGN(-D)
Output: Z<ref> A% -U% ABS(% B% -U% SGN(% +% C -U ABS( D -U SGN( * +%1 =Assign 

Input: Z$ = STR$(A) + STR$(A%)
Output: Z$<ref> A STR$( A% STR$(% +$ =Assign$ 

Input: Z = A% + B * C ^ D
Output: Z<ref> A% B C D ^ * +%1 =Assign 

Input: Z = A$ + B
                ^-- expected string expression

Input: Z = A$ + B%
                ^^-- expected string expression

Input: Z = A + B$
               ^^-- expected numeric expression

Input: Z = A% + B$
                ^^-- expected numeric expression

Input: Z$ = A$ - B$
            ^^-- expected numeric expression

Input: Z$ = MID$(A$,B$,1)
                    ^^-- expected numeric expression

Input: Z$ = MID$(A$,2,C$)
                      ^^-- expected numeric expression

Input: Z$ = MID$(A,2,3)
                 ^-- expected string expression

Input: Z$ = MID$(A$,2,3) + B
                           ^-- expected string expression

Input: Z = B + MID$(A$,2,3)
               ^^^^^^^^^^^^-- expected numeric expression

Input: Z = A + (B$ + C$)
               ^^^^^^^^^-- expected numeric expression

