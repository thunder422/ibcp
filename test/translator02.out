
Input: Z = not A < 5.0 = B > 2.0
Output: Z = NOT A < 5.0 = B > 2.0 

Input: Z% = not A% < 5 and B% > 2 or C% < 1 and D% <> 2
Output: Z% = NOT A% < 5 AND B% > 2 OR C% < 1 AND D% <> 2 

Input: Z$=STR$(VAL("1.23"))
Output: Z$ = STR$(VAL("1.23")) 

Input: Z$=MID$(A$+B$,(C%+5)*D%,4)+" Test"
Output: Z$ = MID$(A$ + B$, (C% + 5) * D%, 4) + " Test" 

Input: Z% = (a$ = "this" + "test")
Output: Z% = (a$ = "this" + "test") 

Input: Z = A% + B
Output: Z = A% + B 

Input: Z = A + B%
Output: Z = A + B% 

Input: Z% = A% + B%
Output: Z% = A% + B% 

Input: Z = A% + B% + C
Output: Z = A% + B% + C 

Input: Z = A% - B
Output: Z = A% - B 

Input: Z = A - B%
Output: Z = A - B% 

Input: Z% = A% - B%
Output: Z% = A% - B% 

Input: Z = A% - B% * C
Output: Z = A% - B% * C 

Input: Z = ABS(A%) + SGN(B%) + ABS(C) * SGN(D)
Output: Z = ABS(A%) + SGN(B%) + ABS(C) * SGN(D) 

Input: Z = ABS(-A%) + SGN(-B%) + ABS(-C) * SGN(-D)
Output: Z = ABS(-A%) + SGN(-B%) + ABS(-C) * SGN(-D) 

Input: Z$ = STR$(A) + STR$(A%)
Output: Z$ = STR$(A) + STR$(A%) 

Input: Z = A% + B * C ^ D
Output: Z = A% + B * C ^ D 

Input: Z = A$ + B
                ^-- expected string expression

Input: Z = A$ + B%
                ^^-- expected string expression

Input: Z = A + B$
               ^^-- expected numeric expression

Input: Z = A% + B$
                ^^-- expected numeric expression

Input: Z$ = A$ - B$
            ^^-- expected numeric expression

Input: Z$ = MID$(A$,B$,1)
                    ^^-- expected numeric expression

Input: Z$ = MID$(A$,2,C$)
                      ^^-- expected numeric expression

Input: Z$ = MID$(A,2,3)
                 ^-- expected string expression

Input: Z$ = MID$(A$,2,3) + B
                           ^-- expected string expression

Input: Z = B + MID$(A$,2,3)
               ^^^^^^^^^^^^-- expected numeric expression

Input: Z = A + (B$ + C$)
               ^^^^^^^^^-- expected numeric expression

